src/core/writer.js

```diff
@@ -77,7 +78,7 @@ function numberToString(value) {
 
 function writeValue(value, buffer, transform) {
   if (isName(value)) {
-    buffer.push(`/${value.name}`);
+    buffer.push(`/${escapePDFName(value.name)}`);
```

Comment:
Could you also add a test in the writer spec so we know that this is in fact called correctly? I think changing https://github.com/mozilla/pdf.js/blob/master/test/unit/writer_spec.js#L68 to a value that needs escaping is already enough since `writeValue` is called by `writeDict` that is being tested there.